pycheribuild/projects/cross/benchmarks.py

```diff
@@ -786,35 +786,40 @@ def install(self, **kwargs):
 
 class NetPerfBench(CrossCompileAutotoolsProject):
     repository = GitRepository("git@github.com:CTSRD-CHERI/cheri-netperf", default_branch="cheri-netperf")
-    cross_install_dir = DefaultInstallDir.CUSTOM_INSTALL_DIR
     target = "netperf"
+    native_install_dir = DefaultInstallDir.IN_BUILD_DIRECTORY
+    cross_install_dir = DefaultInstallDir.ROOTFS_OPTBASE
     # Needs bsd make to build
     make_kind = MakeCommandKind.GnuMake
     # Keep the old bundles when cleaning
     _extra_git_clean_excludes = ["--exclude=*-bundle"]
     # The makefiles here can't support any other other tagets:
-    supported_architectures = [CompilationTargets.CHERIBSD_MIPS_PURECAP, CompilationTargets.CHERIBSD_MIPS_NO_CHERI,
-                               CompilationTargets.CHERIBSD_MIPS_HYBRID, CompilationTargets.NATIVE]
-    _default_install_dir_fn = ComputedDefaultValue(
-        function=lambda c, p: p.bundle_dir,
-        as_string="${BUILDDIR}/netperf-<target-suffix>-bundle")
+    supported_architectures = [CompilationTargets.CHERIBSD_RISCV_NO_CHERI,
+                               CompilationTargets.CHERIBSD_RISCV_HYBRID,
+                               CompilationTargets.CHERIBSD_RISCV_PURECAP]
 
     @classmethod
     def setup_config_options(cls, **kwargs):
         super().setup_config_options(**kwargs)
-
-    @property
-    def bundle_dir(self):
-        return Path(self.build_dir, "netperf-" + self.crosscompile_target.generic_suffix +
-                    self.build_configuration_suffix() + "-bundle")
+        cls.hw_counters = cls.add_config_option("enable-hw-counters",
+                                                choices=("pmc", "statcounters"), default="statcounters",
+                                                help="Use hardware performance counters")
 
     def configure(self, **kwargs):
         if not (self.source_dir / "configure").exists():
             self.run_cmd(self.source_dir / "autogen.sh", cwd=self.source_dir)
         self.configure_args.append("--enable-unixdomain")
+        if self.hw_counters:
+            self.configure_args.append("--enable-pmc={}".format(self.hw_counters))
         self.add_configure_vars(ac_cv_func_setpgrp_void="yes")
         super().configure(**kwargs)
 
+    def process(self):
+        if (self.compiling_for_riscv(include_purecap=True) and
+                self.hw_counters == "pmc"):
+            self.fatal("hwpmc not supported on riscv")
```

Comment:
Is this to support running on Morello? Should we add Morello to the supported architectures?

Reply:
yes and no. Statcounters are not supported but hwpmc should be. I left it out as I have not tested any of it yet, and I have to add the code to netperf to enable and read hwpmc for morello.